src/api/app/controllers/webui/package_controller.rb

```diff
@@ -699,25 +699,18 @@ def check_build_log_access
 
     @package_name = params[:package]
     begin
-      @package = Package.get_by_project_and_name(@project, @package_name, use_source: false,
+      @package = Package.get_by_project_and_name(@project, @package_name, use_source: true,
```

Comment:
With this change from `use_source: false` to `use_source: true`, then rescuing from `Package::ReadAccessError` and `Project::ReadAccessError`, `check_build_log_access` is now like `set_project` and `require_package`. 

See:
https://github.com/openSUSE/open-build-service/blob/eeaaddc878c6c575030c6fff53596762465c621b/src/api/app/controllers/webui/webui_controller.rb#L45-L49

https://github.com/openSUSE/open-build-service/blob/eeaaddc878c6c575030c6fff53596762465c621b/src/api/app/controllers/webui/webui_controller.rb#L129-L150

Reply:
I pushed changes to follow this.